src/cpp/src/whisper/whisper.cpp

```diff
@@ -0,0 +1,201 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include <iostream>
+#include <openvino/openvino.hpp>
+#include <regex>
+#include <thread>
+
+#include "../utils.hpp"
+#include "openvino/genai/streamer_base.hpp"
+#include "openvino/genai/whisper_generation_config.hpp"
+#include "openvino/genai/whisper_speech_recognition_pipeline.hpp"
+#include "whisper_feature_extractor.hpp"
+#include "whisper_models.hpp"
+
+namespace {
+
+void suppress_tokens(ov::Tensor& logits, const std::vector<int64_t>& suppress_tokens) {
+    auto data = logits.data<float>();
+    for (auto supress_token : suppress_tokens) {
+        data[supress_token] = -std::numeric_limits<float>::infinity();
+    }
+}
+
+ov::Tensor encode(ov::InferRequest& request, std::vector<float>& mel_data) {
+    ov::Shape input_shape = {1, 80, 3000};
+    ov::Tensor input_tensor(ov::element::f32, input_shape, mel_data.data());
+
+    request.set_tensor("input_features", input_tensor);
+
+    request.infer();
+
+    // reset input tensor
+    request.set_tensor("input_features", ov::Tensor(ov::element::f32, input_shape));
```

Comment:
it looks expensive to perform allocation just to set dummy tensor

Reply:
Would `request.set_tensor("input_features", ov::Tensor(ov::element::f32, {0, 80, 3000}));` allocate memory?
The current approach from llm pipeline is to set tensor with 0 shape dims: `request.set_tensor("input_ids", ov::Tensor(ov::element::i64, {0, 0}));`. But it doesn't work for whisper as it has fixed dimensions `[?, 80, 3000]`.